pkg/server/plugin/keymanager/awskms/awskms_test.go

```diff
@@ -0,0 +1,706 @@
+			expectedEntries: map[string]keyEntry{
+				spireKeyID: {
+					KMSKeyID: kmsKeyID,
+					PublicKey: &keymanager.PublicKey{
+						Id:   spireKeyID,
+						Type: keymanager.KeyType_RSA_4096,
+					},
+				},
+			},
+		},
+		{
+			name: "missing access key id",
+			configureRequest: ps.configureRequestWith(`{
+				 		"secret_access_key":"secret_access_key",
+				 		"region":"region"
+					 }`),
+			aliases: []types.AliasListEntry{},
+		},
+		{
+			name: "missing secret access key",
+			configureRequest: ps.configureRequestWith(`{
+				 		"access_key_id":"access_key",
+				 		"region":"region"
+					 }`),
+			aliases: []types.AliasListEntry{},
+		},
+		{
+			name: "missing region",
+			configureRequest: ps.configureRequestWith(`{
+				 		"access_key_id":"access_key",
+				 		"secret_access_key":"secret_access_key",
+				 	}`),
+			expectedErr: "awskms: configuration is missing a region",
+		},
+		{
+			name:             "decode error",
+			configureRequest: ps.configureRequestWith("{ malformed json }"),
+			expectedErr:      "awskms: unable to decode configuration: 1:11: illegal char",
+		},
+		{
+			name:             "list aliases error",
+			expectedErr:      "awskms: failed to fetch keys: fake list aliases error",
+			configureRequest: ps.configureRequestWithDefaults(),
+			listAliasesErr:   "fake list aliases error",
+		},
+		{
+			name:             "describe key error",
+			expectedErr:      "awskms: failed to process KMS key: awskms: failed to describe key: describe key error",
+			configureRequest: ps.configureRequestWithDefaults(),
+			aliases: []types.AliasListEntry{
+				{
+					AliasName:   aws.String(kmsAlias),
+					TargetKeyId: aws.String(kmsKeyID),
+				},
+			},
+			describeKeySpec: types.CustomerMasterKeySpecRsa4096,
+			describeKeyErr:  "describe key error",
+		},
+		{
+			name:             "unsupported key error",
+			configureRequest: ps.configureRequestWithDefaults(),
+			aliases: []types.AliasListEntry{
+				{
+					AliasName:   aws.String(kmsAlias),
+					TargetKeyId: aws.String(kmsKeyID),
+				},
+			},
+			describeKeySpec: "unsupported key spec",
+		},
+		{
+			name:             "get public key error",
+			expectedErr:      "awskms: failed to process KMS key: awskms: failed to get public key: get public key error",
+			configureRequest: ps.configureRequestWithDefaults(),
+			aliases: []types.AliasListEntry{
+				{
+					AliasName:   aws.String(kmsAlias),
+					TargetKeyId: aws.String(kmsKeyID),
+				},
+			},
+			describeKeySpec: types.CustomerMasterKeySpecRsa4096,
+			getPublicKeyErr: "get public key error",
+		},
+	} {
+		tt := tt
+		t := ps.T()
+		t.Run(tt.name, func(t *testing.T) {
+			ps.reset()
+			ps.setupListAliases(tt.aliases, tt.listAliasesErr)
+			ps.setupDescribeKey(tt.describeKeySpec, tt.describeKeyErr)
+			ps.setupGetPublicKey(tt.getPublicKeyErr)
+
+			_, err := ps.plugin.Configure(ctx, tt.configureRequest)
+
+			if tt.expectedErr != "" {
+				ps.Require().Error(err)
```

Comment:
Calling assertion/require methods on the test suite from a subtest causes a panic in the test framework. Subtests must use the `*testing.T` that is passed into the callback.